de-lib/src/main/java/org/iplantc/de/server/AppLoggerUtil.java

```diff
@@ -64,6 +64,28 @@ public static AppLoggerUtil getInstance(){
         return request;
     }
 
+    /**
+     * Adds the specified header name and list of IP addresses to the given request, performing validation on
+     * each IP address in the list.
+     *
+     * @param request to which the new header is added.
+     * @param headerName the name of the new header.
+     * @param ipAddresses the list of IP addresses to use as the header value.
+     * @param <T> Basic request object
+     * @return the request with the new header added.
+     */
+    public <T extends HttpRequestBase> T addIpListHeader(T request,
+                                                         final String headerName,
+                                                         final String ipAddresses) {
+        for (String addr : ipAddresses.split("\\s*,\\s*")) {
+            Preconditions.checkArgument(InetAddresses.isInetAddress(addr),
+                                        "Invalid IP address: " + addr);
```

Comment:
I worry a little about stripping off the X-Forwarded-For entirely (or erroring out) if we're passed something that's malformed -- I guess the prior code would already do that though, so maybe this is still an improvement. If we know what's consuming the `X-Forwarded-For` information, would it maybe be possible to push this checking to there, to ensure we're not likely to toss out any useful information (even if possibly malformed)?

Reply:
The request will currently error out if the header is formatted badly. For the time being, I was hoping to modify the code as little as possible while still getting it to work in the Austria deployment. I wouldn't be opposed to making this sort of improvement, but I'd want to take more time to test a bigger change.

Reply:
That sounds sensible to me, thanks for answering.